resolvers.py

```diff
@@ -0,0 +1,1979 @@
+        
+    def process_account_data(self, account_name, account_data):
+        account_projects = []
+        total_pods = 0
+        for project_name, project_data in account_data["children"].items():
+            pods_in_project, project_data_processed = self.process_project_data(
+                project_name, project_data)
+            total_pods = total_pods + pods_in_project
+            account_projects.append(project_data_processed)
+        
+        return {"account_name": account_name, "no_of_projects": len(account_projects), "total_pods": total_pods, "children": account_projects}
+    
+
+class NestingOfWbsTrackingReport(WbsReportNestingHelper):
+    @custom_resolver_check_perm
+    def __call__(self, source, info, **kwargs):
+
+        # Adding Master Filter for billability and Multiple Account Names
+        billability = kwargs.get("billability")
+        filters = kwargs.get("filters")
+        flat_data = models.WbsTrackingFact.objects.all()
+
+        if billability:
+            flat_data = flat_data.filter(billability__in=billability)
+
+        if filters:
+            flat_data = flat_data.filter(account_name__in=filters)
+
+        # Making a Intermediate Grouped Data for our required Nested API
+        grouped_data = self.group_data(flat_data)
+
+        # Nested API
+        nested_data = []
+        for account_name, account_data in grouped_data.items():
+            nested_data.append(self.process_account_data(
+                account_name, account_data))
+
+        return nested_data
+    
+class WBSTrackerAccountsResolver:
+
+    @custom_resolver_check_perm
+    def __call__(self, source, info, **kwargs):
+        qs = models.WbsTrackingFact.objects.all().values('account_name',
+                                                         'billability').distinct('account_name').filter(~Q(billability='Total'))
+        return list(qs)
+
+
+class BillingWBSCodePopUpResolver:
+
+    @custom_resolver_check_perm
+    def __call__(self, source, info, **kwargs):
+        try:
+            pod_id = kwargs.get("pod_id")
+            qs = models.WbsTrackingFact.objects.filter(
+                pod_id=pod_id).values_list('pod_id', 'active_pod_wbs_codes')
+            result = {"pod_id": pod_id}
+            children = []
+            for item in list(qs)[0][1]:
+                value = {}
+                value["wbs_code"] = item["wbs_code"]
+                value["percentage"] = item["allocation_amount_percentage"]
+                children.append(value)
+            result["active_pod_details"] = children
+
+            return result
+
+        except Exception:
+            raise PodIdNotFound("POD-ID does not exist")
+
+
+class DistinctFieldResolver:
+    @custom_resolver_check_perm
+    def __call__(self, source, info, **kwargs):
+        try:
+            model_name = kwargs.get("model_name")
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

You are using the `get` method without a default argument to return the value of a key in a dictionary. We recommended that you use a default argument so that if the value for your key is not found, a default value is returned. If a default value is not provided and the key is not found, then `None` is returned.

[Learn more](https://docs.python.org/3/library/stdtypes.html#dict.get)